code/modules/projectiles/guns/ballistic.dm

```diff
@@ -1,212 +1,207 @@
-		sleep(25)
-		if(user.is_holding(src))
-			process_fire(user, user, 0, zone_override = "head")
-			user.visible_message("<span class='suicide'>[user] blows [user.p_their()] brain[user.p_s()] out with [src]!</span>")
-			return(BRUTELOSS)
-		else
-			user.visible_message("<span class='suicide'>[user] panics and starts choking to death!</span>")
-			return(OXYLOSS)
-	else
-		user.visible_message("<span class='suicide'>[user] is pretending to blow [user.p_their()] brain[user.p_s()] out with [src]! It looks like [user.p_theyre()] trying to commit suicide!</b></span>")
-		playsound(loc, 'sound/weapons/empty.ogg', 50, 1, -1)
-		return (OXYLOSS)
-
-
-
-/obj/item/weapon/gun/projectile/proc/sawoff(mob/user)
-	if(sawn_state == SAWN_OFF)
-		user << "<span class='warning'>\The [src] is already shortened!</span>"
-		return
-	user.changeNext_move(CLICK_CD_MELEE)
-	user.visible_message("[user] begins to shorten \the [src].", "<span class='notice'>You begin to shorten \the [src]...</span>")
-
-	//if there's any live ammo inside the gun, makes it go off
-	if(blow_up(user))
-		user.visible_message("<span class='danger'>\The [src] goes off!</span>", "<span class='danger'>\The [src] goes off in your face!</span>")
-		return
-
-	if(do_after(user, 30, target = src))
-		if(sawn_state == SAWN_OFF)
-			return
-		user.visible_message("[user] shortens \the [src]!", "<span class='notice'>You shorten \the [src].</span>")
-		name = "sawn-off [src.name]"
-		desc = sawn_desc
-		w_class = 3
-		item_state = "gun"
-		slot_flags &= ~SLOT_BACK	//you can't sling it on your back
-		slot_flags |= SLOT_BELT		//but you can wear it on your belt (poorly concealed under a trenchcoat, ideally)
-		sawn_state = SAWN_OFF
-		update_icon()
-		return 1
-
-// Sawing guns related proc
-/obj/item/weapon/gun/projectile/proc/blow_up(mob/user)
-	. = 0
-	for(var/obj/item/ammo_casing/AC in magazine.stored_ammo)
-		if(AC.BB)
-			process_fire(user, user,0)
-			. = 1
-
-
-/obj/item/weapon/suppressor
-	name = "suppressor"
-	desc = "A universal syndicate small-arms suppressor for maximum espionage."
-	icon = 'icons/obj/guns/projectile.dmi'
-	icon_state = "suppressor"
-	w_class = 2
-	var/oldsound = null
-	var/initial_w_class = null
-
-
-/obj/item/weapon/suppressor/specialoffer
-	name = "cheap suppressor"
-	desc = "A foreign knock-off suppressor, it feels flimsy, cheap, and brittle. Still fits all weapons."
-	icon = 'icons/obj/guns/projectile.dmi'
-	icon_state = "suppressor"
-
+/obj/item/weapon/gun/ballistic
+	desc = "Now comes in flavors like GUN. Uses 10mm ammo, for some reason"
+	name = "projectile gun"
+	icon_state = "pistol"
+	origin_tech = "combat=2;materials=2"
+	w_class = 3
+	var/spawnwithmagazine = 1
+	var/mag_type = /obj/item/ammo_box/magazine/m10mm //Removes the need for max_ammo and caliber info
+	var/obj/item/ammo_box/magazine/magazine
+	var/casing_ejector = 1 //whether the gun ejects the chambered casing
```

Comment:
New var so we don't have lots of guns overriding process_chamber() just to change the value of an argument.